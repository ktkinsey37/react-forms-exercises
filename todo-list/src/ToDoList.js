import React, { useState } from "react";
import NewToDoForm from "./NewToDoForm";


function ToDoList() {
  const [todos, setTodos] = useState([]);

  const deleteTodo = (todo) => {
    setTodos(todos.filter(t => t.todo !== todo))
  }

  const renderToDos = () => {
    return (
      <ul>
        {todos.map(todo => (
          <li key={todo.id}>
            {todo.todo} <button onClick={() => deleteTodo(todo.todo)}>X</button>
          </li>
        ))}
      </ul>
    );
  };


  const addToDo = todo => {
    setTodos(todos => [...todos, todo]);
  };
  // end addItem

  return (
    <div className="TodoList">
      <NewToDoForm addToDo={addToDo} />
      {renderToDos()}
    </div>
  );
};
// end

export default ToDoList;
