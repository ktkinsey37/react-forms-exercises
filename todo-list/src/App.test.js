import React from "react";
import { render, fireEvent } from "@testing-library/react";
import ToDoList from './ToDoList';
import App from "./App";

it("renders without crashing", function() {
  render(<App />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

it("can add a new todo and delete it", function() {
  const { getByLabelText, queryByText } = render(<ToDoList />);

  // no items yet
  expect(queryByText("walk the dog")).not.toBeInTheDocument();

  const todoInput = getByLabelText("ToDo:");
  const submitBtn = queryByText("Add a new ToDo!")

  // fill out the form
  fireEvent.change(todoInput, { target: { value: "walk the dog" }});
  fireEvent.click(submitBtn);

  // item exists!
  expect(queryByText("walk the dog")).toBeInTheDocument();

  // delete item

  const deleteTodo = queryByText("X")

  fireEvent.click(deleteTodo)

  expect(queryByText("walk the dog")).not.toBeInTheDocument();

});
