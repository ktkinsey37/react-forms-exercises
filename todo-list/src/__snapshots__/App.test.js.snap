// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="TodoList"
    >
      <form>
        <label
          for="todo"
        >
          ToDo:
        </label>
        <input
          id="todo"
          name="todo"
          placeholder="Add New Todo"
          value="New ToDo"
        />
        <button>
          Add a new ToDo!
        </button>
      </form>
      <ul />
    </div>
  </div>
</DocumentFragment>
`;
